src/pipx/interpreter.py

```diff
@@ -17,6 +21,46 @@ def has_venv() -> bool:
         return False
 
 
+class InterpreterResolutionError(PipxError):
+    def __init__(self, source: str, version: str, wrap_message: bool = True):
+        self.source = source
+        self.version = version
+        potentially_path = "/" in version
+        potentially_pylauncher = "python" not in version and not potentially_path
+
+        message = (
+            f"No executable for the provided Python version '{version}' found in {source}."
+            " Please make sure the provided version is "
+        )
+        if source == "py launcher":
+            message += "listed when running `py --list`."
+        if source == "PATH":
+            message += "on your PATH or the file path is valid. "
+            if potentially_path:
+                message += "The provided version looks like a path, but no executable was found there."
+            if potentially_pylauncher:
+                message += (
+                    "The provided version looks like a version for Python Launcher, " "but `py` was not found on PATH."
+                )
+        super().__init__(message, wrap_message)
+
+
+def find_python_interpreter(python_version: str) -> str:
+    if Path(python_version).is_file():
+        return python_version
+
+    try:
+        py_executable = find_py_launcher_python(python_version)
+        if py_executable:
+            return py_executable
+    except (subprocess.CalledProcessError, FileNotFoundError) as e:
+        raise InterpreterResolutionError(source="py launcher", version=python_version) from e
+
+    if shutil.which(python_version):
+        return python_version
```

Comment:
Hi @Gitznik,
after a little looking into it, I would like to see ``return shutil.which(python_version)`` here.

The reasoning is: I want to be able to give a python executable residing in the path on Windows with e.g. ``--python3.12.exe``. However, since #1168 this is no longer possible, as ``os.path.realpath`` would prepend the cwd and thus, it tries to run ``<current_path>\python3.12.exe``, which doesn't exist, but prevents ``subprocess.run`` from looking up the python executable on the path. (Sidenote: a similar problem was resolved in #1196 for non-Windows platform).

This PR does not change the current situation, however, if the above change would be introduced, the problematic code introduced in #1168 would get a fully resolved path (from ``shutil.which``) and can then perform ``os.,path.realpath`` on that without breaking non-absolute paths.

It remains to say: I personally think, #1168 in this form is a mistake and should never have been merged. Or at least only with much more restrictions, on when ``os.path.realpath`` is executed (e.g. only for abolute paths).

Reply:
Hi guahki,

thanks for your suggestion. While I think it makes sense, I would suggest creating a separate issue for the bug you're describing. Changing the provided python version to a path may or may not have other implications that should be discussed and tested separately.

Reply:
Did that in #1205. Thanks for the quick answer